addon/authenticators/oauth2-password-grant.js

```diff
@@ -324,12 +324,17 @@ export default BaseAuthenticator.extend({
     return new RSVP.Promise((resolve, reject) => {
       fetch(url, options).then((response) => {
         response.text().then((text) => {
-          let json = text ? JSON.parse(text) : {};
+          try {
+            response.responseJSON = text ? JSON.parse(text) : {};
```

Comment:
I'm not sure we should always be modifying `response` like this. Actually modifying it at all was a bad thing to add but we now need to keep it for backwards compatibility. We don't always need to modify though but only when we [reject with the response](https://github.com/simplabs/ember-simple-auth/pull/1442/files#diff-931dc51edd167948e773858f72a47233R335).

Reply:
@RobinDaugherty: what do you think about that? ^

Reply:
Sorry I haven't replied on this. I started making some changes and I ran into some new questions. Please consider this incomplete for now, and I'll respond on this when I can return to it.